pulpcore/content/handler.py

```diff
@@ -957,40 +984,51 @@ def _set_params_from_headers(hdrs, storage_domain):
                         params[STORAGE_RESPONSE_MAP[storage_domain][a_key]] = hdrs[a_key]
             return params
 
+        def _build_url(**kwargs):
+            filename = os.path.basename(content_artifact.relative_path)
+            content_disposition = f"attachment;filename={filename}"
+
+            headers["Content-Disposition"] = content_disposition
+            parameters = _set_params_from_headers(headers, domain.storage_class)
+            storage_url = artifact_file.storage.url(artifact_name, parameters=parameters, **kwargs)
```

Comment:
Since you are refactoring this, how about:
```suggestion
            storage_url = storage.url(artifact_name, parameters=parameters, **kwargs)
```
and then keep the `storage = domain.get_storage()` line where it was.

Reply:
Done! Even though we now run on a bit more CPU ticks.